web/src/components/DateField.js

```diff
@@ -19,9 +20,58 @@ const joinDate = ({ day, month, year }) => `${year}-${month}-${day}`;
 const DateField = ({ value, onChange, ...rest }) => {
   const dateParts = splitDate(value);
 
+  const errorInfo = useMemo(() => {
+    const date = moment(value, 'YYYY-M-D', true);
+    if (date.isValid()) {
+      return null;
+    }
+
+    const message = [];
+    let dayInvalid = false;
+    let monthInvalid = false;
+    let yearInvalid = false;
+
+    if (!dateParts.month) {
+      message.push('Month is required.');
+      monthInvalid = true;
+    }
+    if (dateParts.month > 12) {
+      message.push('Month must be between 1 and 12.');
+      monthInvalid = true;
+    }
+
+    if (!dateParts.day) {
+      message.push('Day is required.');
+      dayInvalid = true;
+    }
+    if (dateParts.day > 31) {
+      message.push('Day must be less than 31.');
+      dayInvalid = true;
+    }
+
+    if (`${dateParts.year}`.length !== 4) {
+      message.push('Year must be 4 digits.');
+      yearInvalid = true;
+    }
+
+    if (message.length === 0) {
+      message.push(
+        'Invalid date - is the day number too high for the provided month and year?'
```

Comment:
Is this the only circumstance we know of that would cause an invalid date?

Reply:
It's the only one I'm aware of that isn't covered by the other checks.  I'm sure users will find others!